src/Unused/CLI/SearchWithProgress.hs

```diff
@@ -0,0 +1,43 @@
+module Unused.CLI.SearchWithProgress
+    ( searchWithProgressBar
+    ) where
+
+import Control.Monad.State
+import System.IO (hSetBuffering, BufferMode(NoBuffering), stdout)
+import System.ProgressBar
+import Unused.TermSearch (search)
+
+searchWithProgressBar :: [String] -> IO [String]
+searchWithProgressBar terms = do
+    (results, _) <- runStateT (performSearch $ length terms) terms
+    return $ concat results
+
+performSearch :: Int -> StateT [String] IO [[String]]
+performSearch total = do
+    currentTerm <- getSearchTerm
+    searchResults <- liftIO $ search currentTerm
+
+    remainingTerms <- get
+    let remaining = length remainingTerms
+
+    liftIO $ printProgressBar (total - remaining) total
+
+    if remaining > 0
+        then do
+            res <- performSearch total
+            return $ searchResults:res
+        else return [searchResults]
+
+getSearchTerm :: StateT [String] IO String
+getSearchTerm = do
+    (x:xs) <- get
+    put xs
+    return x
+
+printProgressBar :: Int -> Int -> IO ()
+printProgressBar complete total = do
+    let message = "Working"
+    let progressBarWidth = 60
+
+    hSetBuffering stdout NoBuffering
```

Comment:
what about setting this in `main` or more toward the beginning of your app? That way you're not setting it ever time you print